File: common/common-domain/src/main/java/com/kittens/exploding/common/base_domain/Card.java
package com.kittens.exploding.common.base_domain;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
public class Card {
    private final String id;
    // private final Texture texture;
    private final CardCategory category;
    private final String personalityCaption;
    private final String jukeboxId;

    private boolean isDiarrhead;
    private boolean isDisguisedAsCatButt;
    private boolean isMarked;
    private DrawType drawType;
    // private Player placingPlayer;

    private CardEvent onSprayedWithDiarrhea;
    private CardEvent onDiarrheaWiped;
    private CardEvent onDisguised;
    private CardEvent onSheddedDisguise;
    private CardEvent onMarked;
    private CardEvent onUnMarked;

    public boolean hasPlacingPlayer() {
        return false;
    }

    public boolean hasJukeboxSound() {
        return false;
    }

    public boolean hasDiscardCardSubstituteCategory() {
        return false;
    }

    public CardCategory discardCardSubstituteCategory() {
        return null;
    }

    public boolean restartsTurnTimerWhenPlayed() {
        return false;
    }

    // public CategoryMetadata categoryMetadata() {
    //     return null;
    // }

    public boolean isValidInStartingHand() {
        return false;
    }

    public boolean isExplodingKitten() {
        return false;
    }

    public boolean isDefuse() {
        return false;
    }

    public boolean isSelfSlap() {
        return false;
    }

    public void reset() {}
    public void markCard() {}
    public void unMarkCard() {}
    public void sprayWithDiarrhea() {}
    public void wipeUpDiarrhea() {}
    public void disguise() {}
    public void shedDisguise() {}
}

File: Kittens_5.3.11_project/Scripts/Assembly-CSharp/Card.cs
using System;
using UnityEngine;

[Serializable]
public class Card
{
	public string Id;

	public Texture Texture;

	public CardCategory Category;

	public string PersonalityCaption;

	public string JukeboxId;

	public bool IsDiarrhead;

	public bool IsDisguisedAsCatButt;

	public bool IsMarked;

	public DrawType DrawType;

	[HideInInspector]
	public Player PlacingPlayer;

	public CardEvent OnSprayedWithDiarrhea;

	public CardEvent OnDiarrheaWiped;

	public CardEvent OnDisguised;

	public CardEvent OnSheddedDisguise;

	public CardEvent OnMarked;

	public CardEvent OnUnMarked;

	public bool HasPlacingPlayer => false;

	public bool HasJukeboxSound => false;

	public bool HasDiscardCardSubstituteCategory => false;

	public CardCategory DiscardCardSubstituteCategory => default(CardCategory);

	public bool RestartsTurnTimerWhenPlayed => false;

	public CategoryMetadata CategoryMetadata => null;

	public bool IsValidInStartingHand => false;

	public bool IsExplodingKitten => false;

	public bool IsDefuse => false;

	public bool IsSelfSlap => false;

	public Card(string id, CardCategory category, Texture texture, string jukeboxId, string personalityCaption)
	{
	}

	public void Reset()
	{
	}

	public void MarkCard()
	{
	}

	public void UnMarkCard()
	{
	}

	public void SprayWithDiarrhea()
	{
	}

	public void WipeUpDiarrhea()
	{
	}

	public void Disguise()
	{
	}

	public void ShedDisguise()
	{
	}
}
